File: 7KYU/Speed Control/solution.clj
(ns gps.core)
(defn gps [s x]
  (println (count x))
  (loop [max 0.0 r (rest x) l (first x)]
    (if (empty? r)
      (int (Math/floor max))
      (let [n (* 60.0 60.0 (/ (- (first r) (float l)) s))]
        (if (> n max)
          (recur n (rest r) (first r))
          (recur max (rest r) (first r)))))))

File: 7KYU/Speed Control/solution.cs
using System;
using System.Linq;
public class GpsSpeed {
    
    public static int Gps(int s, double[] x)
    {
        var max = 0.0;
        for(int i = 1; i < x.Length; i++)
        {
          var cmax = ((x[i] - x[i-1])/s) * 3600;
          if(cmax > max)
            max = cmax;
        }
        return (int)Math.Floor(max);
    }
}